include/boost/compute/utility/program_cache.hpp

```diff
@@ -82,6 +82,35 @@ class program_cache : boost::noncopyable
         m_cache.clear();
     }
 
+    /// Clears persistent cache by deleting the boost_compute folder
+    static void clear_persistent_cache()
+    {
+#ifdef BOOST_COMPUTE_USE_OFFLINE_CACHE
+        // Path delimiter symbol for the current OS.
+        static const std::string delim = boost::filesystem::path("/").make_preferred().string();
```

Comment:
So instead of doing all this here, you should call the [`detail::appdath_path()`](https://github.com/kylelutz/compute/blob/develop/include/boost/compute/detail/path.hpp#L30) function defined in `<boost/compute/detail/path.hpp>`. I recently extracted this code from the `program` class so you may have to rebase on top of the `develop` branch to get it.